src/Zicht/Bundle/AdminBundle/DataTransformer/ClassTransformer.php

```diff
@@ -43,10 +44,14 @@ public function __construct(Quicklist $lister, $repo)
      */
     public function transform($value)
     {
-        return array(
-            'id' => (null !== $value ? $value->getId() : null),
-            'value' => (null !== $value ? (string)$value : null)
-        );
+        try {
+            return array(
+                'id' => (null !== $value ? $value->getId() : null),
+                'value' => (null !== $value ? $value->__toString() : null)
+            );
+        } catch (EntityNotFoundException $e) {
+            return ['id' => null, 'value' => '-- ENTITY NOT FOUND --'];
```

Comment:
It's clearer for the user that an entry can't be found (this was an issue with the versioning-bundle when trying to transform an entity that didn't existed anymore)
After persisting the value is set to null, so it's only shown once.